### PR TITLE
Fix nxp 28119 pdf extracttext with empty searchpattern

### PR DESCRIPTION
**Pull request by ThibArg**

### CODE REVIEW

**Comment by ataillefer** [comment]
I have a slight difference when formatting the test class.
What IDE/version are you using? Is your formatting configuration up-to-date?

**Comment by ataillefer** [comment]
You can squash https://github.com/nuxeo/nuxeo/pull/3447/commits/b312405454f250c7af9375fd98c00aed8f1af71f and https://github.com/nuxeo/nuxeo/pull/3447/commits/60d9f212fb6f2b99061b045dd62292c0c57281f7.

**Comment by ataillefer** [comment]
Just wondering why `PDF_ExtractText` in the commit messages instead of `PDF.ExtractText` or `PDFExtractTextOperation`?

File: addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java [hunk 1]
```diff
@@ -93,17 +93,54 @@ public void tearDown() {
...
+        chain.add(PDFExtractTextOperation.ID)
+             .set("save", false)
+             .set("targetxpath", "dc:description")
+             .set("patterntofind", "");
+        documentModified = (DocumentModel) automationService.run(ctx, chain);
+        text = (String) documentModified.getPropertyValue("dc:description");
+        assertNotNull(text);
+        assertEquals(19045, text.length());
+        assertFalse(text.contains("Something that is not in the file"));
+        assertTrue(text.contains("13.1"));
```
Comment (ataillefer) [inline addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java]: You could factorize the 2 cases as the only difference is setting `patterntofind` or not.

File: addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java [hunk 2]
```diff
@@ -113,22 +150,26 @@ public void testExtractTextOperation() throws Exception {
...
         chain.add(PDFExtractTextOperation.ID)
              .set("save", true)
              .set("targetxpath", "dc:description")
-             .set("patterntofind", "Contract Number: ")
-             .set("removepatternfromresult", true);
+             .set("patterntofind", "Something that is not in the file")
+             .set("removepatternfromresult", false);
         documentModified = (DocumentModel) automationService.run(ctx, chain);
-        assertEquals("123456789", documentModified.getPropertyValue("dc:description"));
+        assertNull(documentModified.getPropertyValue("dc:description"));
```
Comment (ataillefer) [inline addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java]: You could also factorize the 3 cases, though that's just improving some existing code.

File: addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java (hunk 1 above)
Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java]: Ok, done in next commits

File: addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java (hunk 2 above)
Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/test/java/org/nuxeo/ecm/platform/pdf/tests/PDFTextExtractorTest.java]: Ok, done in next commit(s)

File: addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java [hunk 3]
```diff
@@ -39,8 +41,9 @@
  *
  * @since 8.10
  */
-@Operation(id = PDFExtractTextOperation.ID, category = Constants.CAT_DOCUMENT, label = "PDF: Extract Text",
-    description = "Extracts raw text from a PDF. If the PDF is encrypted, a password is required.")
+@Operation(id = PDFExtractTextOperation.ID, category = Constants.CAT_DOCUMENT, label = "PDF: Extract Text", description = "Extracts raw text from a PDF."
+        + " If the PDF is encrypted, a password is required."
+        + " If patterntofind is not provided, extracts all the text it can.")
```
Comment (efge) [inline addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: While you're editing this, could you add something like `The extracted text is then set in the targetxpath property of the input document`? That's not obvious and quite important I would think. You could also mention the removepatternfromresult property maybe...

Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: ok, doing it

Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: (explaining all parameters)

Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: (done. I will squash once approved)

File: addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java [hunk 4]
```diff
@@ -70,8 +76,13 @@
...
         textExtractor.setPassword(password);
-        String extractedText = removepatternfromresult ?
-            textExtractor.extractLastPartOfLine(patterntofind) : textExtractor.extractLineOf(patterntofind);
+        String extractedText;
+        if (StringUtils.isBlank(patterntofind)) {
+            extractedText = textExtractor.getAllExtractedLines();
+        } else {
+            extractedText = removepatternfromresult ? textExtractor.extractLastPartOfLine(patterntofind)
+                    : textExtractor.extractLineOf(patterntofind);
+        }
```
Comment (efge) [inline addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: Sorry last one. For readability could you write this as:

        if (StringUtils.isBlank(patterntofind)) {
             extractedText = textExtractor.getAllExtractedLines();
         } else if (removepatternfromresult) {
             extractedText = textExtractor.extractLastPartOfLine(patterntofind);
         } else {
             extractedText = textExtractor.extractLineOf(patterntofind);
         }

Comment (ThibArg) [reply addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: Yes I can :-) (done)

Comment (efge) [reply addons/nuxeo-platform-pdf-utils/src/main/java/org/nuxeo/ecm/platform/pdf/operations/PDFExtractTextOperation.java]: Thanks

**Pull request merged**